notebook/services/kernels/kernelmanager.py

```diff
@@ -463,3 +467,38 @@ def cull_kernel_if_idle(self, kernel_id):
                 self.log.warning("Culling '%s' kernel '%s' (%s) with %d connections due to %s seconds of inactivity.",
                                  kernel.execution_state, kernel.kernel_name, kernel_id, connections, idle_duration)
                 self.shutdown_kernel(kernel_id)
+
+    @gen.coroutine
+    def kernel_ram_monitoring(self):
+        # Arbitrary time for waiting till next metric collection
+        yield gen.sleep(0.2)
+
+        # Update the relevant kernel's ram usage metric
+        def update_ram_metric():
+            pid = kernel.kernel.pid
+            kernel_process = psutil.Process(pid)
```

Comment:
This is assuming that all kernels are local to the server - which is not the case for Enterprise Gateway - which derives its kernel management from these classes.  In this case, EG will break since the launching process is no longer running.

I would recommend moving the code that gets the memory usage to a method on the individual kernel manager class - this way child classes can do their own means of capturing memory usage.  Unfortunately, Notebook doesn't define a kernel manager class.  If one was added, it would derive from jupyer_client's [`IOLoopKernelManager`](https://github.com/jupyter/jupyter_client/blob/master/jupyter_client/ioloop/manager.py#L26) - then we'd need to fix up the hierarchy usage in Jupyter Kernel and Enterprise Gateway.  I'd be happy to help with this if others agree.  The other option is to abstract the `kernel` class that represents the process.  We do this in Enterprise Gateway with our `process proxy` classes, but that's a bit more work.  We'd then have a 'hook' to place capture code for our remote kernels.

Any way we can make the capture of per-kernel info more abstract would be great.

In the meantime, we should probably make this feature optional and, preferably, off by default so as to not break applications.